x/ecocredit/server/testsuite/suite.go

```diff
@@ -110,178 +107,179 @@ func (s *IntegrationTestSuite) SetupSuite() {
-	require.Equal(tRes.Credits[0].Amount, creditAmtDeposited.String())
-
-	// user shouldn't be able to take any since we sent our tokens to user2
-	noRes, err := s.basketServer.Take(s.ctx, &basket.MsgTake{
-		Owner:              user.String(),
-		BasketDenom:        basketDenom,
-		Amount:             basketTokensReceived.String(),
-		RetirementLocation: "US-NY",
-		RetireOnTake:       false,
-	})
-	require.Error(err)
-	require.Contains(err.Error(), sdkerrors.ErrInsufficientFunds.Error())
-	require.Nil(noRes)
-
-	// there should be nothing left in the basket
-	bRes, err := s.basketServer.BasketBalance(s.ctx, &basket.QueryBasketBalanceRequest{
-		BasketDenom: basketDenom,
-		BatchDenom:  batchDenom,
-	})
-	require.Error(err)
-	require.Contains(err.Error(), "not found")
-	require.Nil(bRes)
-
-	// basket token balance of user2 should be empty now
-	endBal := s.getUserBalance(user2, basketDenom)
-	require.True(endBal.Amount.Equal(sdk.NewInt(0)), "ending balance was %s, expected 0", endBal.Amount.String())
-
-	// check retire enabled basket
-
-	s.mockDist.EXPECT().FundCommunityPool(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(interface{}, interface{}, interface{}) error {
-		err := s.bankKeeper.SendCoinsFromAccountToModule(s.sdkCtx, user, ecocredit.ModuleName, sdk.NewCoins(s.basketFee))
-		return err
-	})
-	// create a retire enabled basket
-	resR, err := s.basketServer.Create(s.ctx, &basket.MsgCreate{
-		Curator:           s.signers[0].String(),
-		Name:              "RETIRE",
-		Exponent:          6,
-		DisableAutoRetire: false,
-		CreditTypeAbbrev:  "BAZ",
-		AllowedClasses:    []string{classId},
-		DateCriteria:      nil,
-		Fee:               sdk.NewCoins(s.basketFee),
-	})
-	require.NoError(err)
-	basketDenom = resR.BasketDenom
-
-	creditsToDeposit := math.NewDecFromInt64(3)
-
-	// put some credits in the basket
-	pRes, err = s.basketServer.Put(s.ctx, &basket.MsgPut{
-		Owner:       user.String(),
-		BasketDenom: basketDenom,
-		Credits:     []*basket.BasketCredit{{Amount: creditsToDeposit.String(), BatchDenom: batchDenom}},
-	})
-	require.NoError(err)
-
-	amountBasketCoins, err := math.NewDecFromString(pRes.AmountReceived)
-	require.NoError(err)
-
-	// take them out of the basket, retiring them
-	tRes, err = s.basketServer.Take(s.ctx, &basket.MsgTake{
-		Owner:              user.String(),
-		BasketDenom:        basketDenom,
-		Amount:             amountBasketCoins.String(),
-		RetirementLocation: "US-NY",
-		RetireOnTake:       true,
-	})
-	require.NoError(err)
-	require.Len(tRes.Credits, 1) // should only be one credit
-	require.Equal(creditsToDeposit.String(), tRes.Credits[0].Amount)
-
-	// check retired balance, should be equal to the amount we put in
-	cbRes, err := s.queryClient.Balance(s.ctx, &ecocredit.QueryBalanceRequest{
-		Account:    user.String(),
-		BatchDenom: batchDenom,
-	})
-	require.NoError(err)
-	require.Equal(creditsToDeposit.String(), cbRes.RetiredAmount)
-}
+// TODO: reimpl after full submodule integration
```

Comment:
What items are missing before full submodule integration?

Reply:
params, ops/sims. this test was failing since i started using core.Params, and the integration test/app is wired up for v1alpha params

Reply:
As mentioned above, I think params should remain in the root level as it includes params for both `core` and `basket`. I guess this is what you were referring to as "refactor tests" in #728.

Reply:
This is not ideal but we can handle in a separate pull request when we refactor tests #728.